src/SonarQube.Client/Services/SonarQubeClient.cs

```diff
@@ -225,22 +225,28 @@ public Task<Result<VersionResponse>> GetVersionAsync(CancellationToken token)
         public Task<Result<CredentialResponse>> ValidateCredentialsAsync(CancellationToken token)
         {
             // Since 3.3
-            var validateCredentialsApiUrl = new Uri("api/authentication/validate", UriKind.Relative);
+            var validateCredentialsApiUrl = BuildRelativeUrl("api/authentication/validate");
 
             return InvokeSonarQubeApi(validateCredentialsApiUrl, token,
                 stringResponse => new CredentialResponse { IsValid = (bool)JObject.Parse(stringResponse).SelectToken("valid") });
         }
 
-        private Uri BuildRelativeUrl(string relativePath, Dictionary<string, string> queryParameters)
+        private Uri BuildRelativeUrl(string relativePath, Dictionary<string, string> queryParameters = null)
         {
             var query = HttpUtility.ParseQueryString(string.Empty);
-            foreach (var kvp in queryParameters)
+
+            foreach (var kvp in queryParameters ?? new Dictionary<string, string>()
+                .Where(p => p.Value != null))
```

Comment:
Wow...Are you sure the `.Where` applies to both and not only the `new Dictionary`? Even if it does, I would extract this into a separate variable because it is not easy to read.